superset/models.py

```diff
@@ -1346,13 +1352,27 @@ def visit_column(element, compiler, **kw):
 
             dttm_col = cols[granularity]
             time_grain = extras.get('time_grain_sqla')
+            time_filters = []
 
-            if is_timeseries:
+            # If primarily grouping by time without join, place time groupings first
+            if is_timeseries and time_groupby_inline:
+                timestamp = dttm_col.get_timestamp_expression(time_grain)
+                select_exprs = [timestamp] + select_exprs
+                groupby_exprs = [timestamp] + groupby_exprs
+            elif is_timeseries and not time_groupby_inline:
                 timestamp = dttm_col.get_timestamp_expression(time_grain)
                 select_exprs += [timestamp]
                 groupby_exprs += [timestamp]
 
-            time_filter = dttm_col.get_time_filter(from_dttm, to_dttm)
+            # Use all dttm columns to support index with secondary dttm columns
+            dttm_col_all = False
+            if hasattr(self.database.db_engine_spec, 'time_secondary_columns'):
```

Comment:
Now you don't need the `hasattr` anymore and perhaps not `dttm_col_all`, just use time_secondary_columns